docs/data/data-grid/master-detail/master-detail.md

```diff
@@ -1,43 +1,63 @@
 ---
-title: Data Grid - Master detail
+title: Data Grid - Master-detail row panels
 ---
 
-# Data Grid - Master detail [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
+# Data Grid - Master-detail row panels [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
 
-<p class="description">Expand your rows to display additional information.</p>
+<p class="description">Implement master-detail row panels to let users view extended information without leaving the Data Grid.</p>
 
-The master detail feature allows expanding a row to display additional information inside a panel.
-To use this feature, pass a function to the `getDetailPanelContent` prop with the content to be rendered inside the panel.
-Any valid React element can be used as the row detail, even another grid.
+## What is the master-detail pattern?
+
+"Master-detail" is a design pattern for organizing information in which the "master" area lists the data and the "detail" sections provide further information about each item.
+
+The Data Grid Pro provides the tools to implement master-detail row panels.
+These are useful whenever you need to give end users additional information about row items without navigating away from the Grid.
+
+A common example of this pattern is found in many email clients—users can click on an email from the master list to see its contents (details) as well as actions they can take such as replying, forwarding, and deleting.
+To expand a row, click on the **+** icon or press <kbd class="key">Space</kbd> when focused on a cell in the detail toggle column:
+
+{{"demo": "MasterDetailEmailExample.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Implementing master-detail row panels
+
+To create master-detail row panels, pass a function to the `getDetailPanelContent` prop with the content to be rendered inside the panel.
+You can use any valid React element—even another Data Grid.
+
+```tsx
+<DataGridPro getDetailPanelContent={({ row }) => <div>Row ID: {row.id}</div>} />
+```
+
+### Detail panel height
 
 By default, the detail panel height is 500px.
-You can customize it by passing a function to the `getDetailPanelHeight` prop.
-This function must return either a number or the `"auto"` string.
-If it returns a number, then the panel will use that value (in pixels) for the height.
-If it returns `"auto"`, then the height will be [derived](#infer-height-from-the-content) from the content.
+You can customize this by passing a function to the `getDetailPanelHeight` prop.
+This function must return either a number or `"auto"`:
+
+- If it returns a number, then the panel will use that value (in pixels) for the height.
+- If it returns `"auto"`, then the height will be derived from the content.
```

Comment:
if we ignore the vale runner on some keywords anyways, wouldn't it make sense to disable that keyword then?